diff --git a/helm-taskrunner.el b/helm-taskrunner.el
index f81eb80..24a795c 100644
--- a/helm-taskrunner.el
+++ b/helm-taskrunner.el
@@ -5,7 +5,7 @@
 ;; Author: Yavor Konstantinov <ykonstantinov1 AT gmail DOT com>
 ;; URL: https://github.com/emacs-taskrunner/helm-taskrunner
 ;; Version: 1.0
-;; Package-Requires: ((emacs "25.1") (projectile "2.0.0") (helm "v3.0"))
+;; Package-Requires: ((emacs "25.1") (projectile "2.0.0") (helm "3.0"))
 ;; Keywords: build-system taskrunner build task-runner tasks helm convenience
 
 ;; This file is not part of GNU Emacs.
